features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/data/AuthManager.js

```diff
@@ -172,14 +169,25 @@ class AuthManager {
             validity_period: 3600,
             scopes: 'apim:api_view apim:api_create apim:api_publish apim:tier_view apim:tier_manage apim:subscription_view apim:subscription_block apim:subscribe'
         };
-        let promised_response = axios.post(this.getTokenEndpoint(environment), qs.stringify(data), {headers: headers});
+        let promised_response = axios(this.getTokenEndpoint(environment), {
+            method: "POST",
+            data: qs.stringify(data),
+            headers: headers,
+            withCredentials: true
+        });
+        //Set the environment that user tried to authenticate
+        let previous_environment = Utils.getEnvironment();
+        Utils.setEnvironment(environment);
+
         promised_response.then(response => {
             const validityPeriod = response.data.validityPeriod; // In seconds
             const WSO2_AM_TOKEN_1 = response.data.partialToken;
             const user = new User(response.data.authUser, response.data.idToken);
-            user.setPartialToken(WSO2_AM_TOKEN_1, validityPeriod, this.contextPath);
+            user.setPartialToken(WSO2_AM_TOKEN_1, validityPeriod, Utils.CONST.CONTEXT_PATH);
             user.scopes = response.data.scopes.split(" ");
             AuthManager.setUser(user);
+        }).catch(error => {
+            Utils.setEnvironment(previous_environment);
```

Comment:
Shall we log an error message here.

Reply:
Fixed with the commit https://github.com/RenukaFernando/carbon-apimgt/commit/f23f437254d2faae448a18be410a07137fb7689b